### ISSUE TITLE
Scons build fails at c++ compiler check.

### ISSUE THREAD

**Issue by JureBericic** [JeffersonLab/analyzer#97]
Cannot build the analyzer using the `scons`. It fails with:
```
    Got a 64-bit processor, I can do a 64-bit build in theory...
    64-bit Linux build
    Checking whether the C++ compiler works... no
    !!! Your compiler and/or environment is not correctly configured.
```

However, the build using `make` works fine.

**Comment by brash99** [comment]
Hi Jure,

As the message says, there appears to be something about your C++ environment that is not configured the way that scons required (which admittedly is not very much).  Can you send me (brash@jlab.org) the file config.log that is produced when you run scons?

Cheers,
E.

Dr. Edward J. Brash

Professor of Physics - Christopher Newport University
Staff Scientist - Thomas Jefferson National Accelerator Facility
Honorary Senior Research Fellow - University of Glasgow
Office:	757-594-7451
Mobile:	757-753-2831
FAX:		757-594-7919

**Comment by hansenjo** [comment]
Confirmed on my desktop machine. Same message. I'm using RHEL 7 and scons 2.5.0 from EPEL. The scons package was last updated in August of this year. Earlier this year, scons builds worked just fine on my box. Nothing obvious in the compiler environment has changed since then.

**Comment by brash99** [comment]
I’ll figure out what is going on.

**Comment by brash99** [comment]
Ole, can you tell me what compiler version your RHEL 7 system is using, and also could you post the config.log file that is created when you type scons?

I have tested with an EL6 system and scons2.5.0 and it works ok for me.  I will create an EL7 virtual machine tonight and try to recreate the behavior.  I tried initially to use ifarm1402 (which is Centos7.2) at jlab, but it looks like they don’t have ROOT available for it yet … argh!

Cheers,
E.

**Comment by hansenjo** [comment]
Hi Ed,

the compiler version:

$ gcc --version
gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-4)

and config.log:

$ cat config.log
file /home/ole/Develop/analyzer/SConstruct,line 163:
	Configure(confdir = .sconf_temp)
scons: Configure: Checking whether the C++ compiler works...
scons: Configure: ".sconf_temp/conftest_0.cpp" is up to date.
scons: Configure: The original builder output was:
  |.sconf_temp/conftest_0.cpp <-
  |  |
  |  |int main()
  |  |{
  |  |    return 0;
  |  |}
  |  |
  |
c++ -o .sconf_temp/conftest_0.o -c -std=c++11 -O "-Wall
-Woverloaded-virtual" -Wextra -Wno-missing-field-initializers
-Wno-unused-parameter -Wno-deprecated-declarations -m64 -pthread
-rdynamic -DNDEBUG -DLINUXVERS -DHAS_SSTREAM
-I/home/ole/evio-4.4.6/Linux-x86_64/include -Isrc -Ihana_decode
-I/usr/include/root .sconf_temp/conftest_0.cpp
c++: error: unrecognized command line option ‘-Wall -Woverloaded-virtual’
scons: Configure: no

Ahh ... I think we're on to something here.

Thanks for looking into this.

Ole

**Comment by brash99** [comment]
Interesting.

On ifarm1402, scons uses g++ rather than c++, but c++ and g++ are the same exact executable.  Is this true on your RHEL7 system?  What are the outputs of c++ —version and g++ —version?

Cheers,
E.

**Comment by hansenjo** [comment]
Closed by commit df29d2b569322c80f4ce2021c58af67a57a9ded6
